src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchProcessResponseFilesImpl.java

```diff
@@ -0,0 +1,182 @@
+        if (osrOptional.isEmpty()) {
+            log.error("Object State Record not found for Arm EOD {}", armEod);
+        }
+        return osrOptional;
+    }
+
+    private void updateOsrResponseReceivedAttributes(ObjectStateRecordEntity osr) {
+        osr.setFlagRspnRecvdFromArml(true);
+        osr.setDateRspnRecvdFromArml(timeHelper.currentOffsetDateTime());
+        osrRepository.save(osr);
+    }
+
+    private void deleteBlobDataAndEod(ExternalObjectDirectoryEntity armEod, ObjectStateRecordEntity osr) {
+        List<ExternalObjectDirectoryEntity> detsEods = externalObjectDirectoryRepository.findExternalObjectDirectoryByLocation(
+            EodHelper.detsLocation(),
+            armEod.getMedia(),
+            armEod.getTranscriptionDocumentEntity(),
+            armEod.getAnnotationDocumentEntity(),
+            armEod.getCaseDocument());
+
+        if (detsEods.size() == 1) {
+            boolean deleted = detsDataStoreDeleter.delete(detsEods.getFirst());
+            if (deleted) {
+                osr.setFlagFileDetsCleanupStatus(true);
+                osr.setDateFileDetsCleanup(timeHelper.currentOffsetDateTime());
+                osrRepository.save(osr);
+            } else {
+                String errorMessage = String.format("Unable to delete DETS blob for ARM EDO %s", armEod.getId());
+                log.error(errorMessage);
+                updateOsrIngestStatusToFailure(osr, errorMessage);
+            }
+        } else {
+            String errorMessage = String.format("Unable to delete DETS blob because either no DETS EOD for ARM EDO %s found or more than one DETS EOD found", armEod.getId());
+            log.error(errorMessage);
+            updateOsrIngestStatusToFailure(osr, errorMessage);
+        }
+    }
+
+    private void updateOsrFileIngestStatusToSuccess(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor,
+                                                    ArmResponseBatchData armResponseBatchData,
+                                                    String objectChecksum,
+                                                    ObjectStateRecordEntity osr) {
+        osr.setFlagFileIngestStatus(true);
+        osr.setDateFileIngestToArm(timeHelper.currentOffsetDateTime());
+        osr.setMd5FileIngestToArm(objectChecksum);
+        osr.setIdResponseFile(batchUploadFileFilenameProcessor.getInputUploadFilename());
+        osr.setIdResponseCrFile(armResponseBatchData.getCreateRecordFilenameProcessor().getCreateRecordFilename());
+        osr.setIdResponseUfFile(armResponseBatchData.getUploadFileFilenameProcessor().getUploadFileFilename());
+        osrRepository.save(osr);
+    }
+
+    private void updateOsrIngestStatusToFailure(ObjectStateRecordEntity osr,
+                                                String objectStatus) {
+        osr.setFlagFileIngestStatus(false);
+        osr.setDateFileIngestToArm(timeHelper.currentOffsetDateTime());
+        osr.setObjectStatus(objectStatus);
```

Comment:
Do we want this field to be appended to so if there are any previous failures, the failures are maintained?

Reply:
@hemantasharma1129 ?

Reply:
Ideally, when an exception occurs we should stop right there and move to the next execution. But if there are sequence of catch blocks during an exception and we are updating the OSR status in each catch block then its a good idea to append the Previous failure.

Reply:
Ok will implement the append then 👍

Reply:
By having a better look at the DETS flow I've realised that the above is not necessary, because all the steps where we update the `objectStatus` field are final steps. i.e. for eache execution of the DETS flow the `objectStatus` will only be updated only once. Proven by existing integration tests. 
I will apply the YAGNI principle and skip this change for now. If needed in the future this can be introduced

Reply:
I see, agree. this is not required then.

Reply:
If it fails and the runs again this field will still have a value. Should this field be cleared out if successful?

Reply:
that's a good point, added